MinecraftClient/McTcpClient.cs

```diff
@@ -1664,5 +1664,20 @@ public void OnHeldItemChange(byte slot)
                 bot.OnHeldItemChange(slot);
             CurrentSlot = slot;
         }
+        
+        public void ChangeGamemode(int gamemode, Guid uuid)
+        {
+            string playerName = null;
+            if (onlinePlayers.ContainsKey(uuid))
+            {
+                playerName = onlinePlayers[uuid];
+                this.gamemode = gamemode;

```

Comment:
Since this event is changing some player gamemode it's not necessarily the client's gamemode. Maybe a check is missing here?